pkg/controller/rmdworkload/rmdworkload_controller_test.go

```diff
@@ -50,14 +50,57 @@ func createReconcileRmdWorkloadObject(rmdWorkload *intelv1alpha1.RmdWorkload) (*
 
 }
 
+func createListeners(address string, httpResponses map[string]([]rmdtypes.RDTWorkLoad)) (*httptest.Server, error) {
+	var err error
+
+	// create a listener with the desired port.
+	newListener, err := net.Listen("tcp", address)
+	if err != nil {
+		fmt.Errorf("Failed to create listener")
+	}
+
+	//create muxes to handle get requests
+	mux := http.NewServeMux()
+	mux.HandleFunc("/v1/workloads/", (func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "GET" {
+			b, err := json.Marshal(httpResponses[address])
+			if err == nil {
+				fmt.Fprintln(w, string(b[:]))
+			}
+		} else if r.Method == "POST" {
+			if err == nil {
+				fmt.Fprintln(w, "successful post")
+			}
+		} else if r.Method == "PATCH" {
+			if err == nil {
+				fmt.Fprintln(w, "OK")
+			}
+		} else {
+			if err == nil {
+				fmt.Fprintln(w, "ok")
+			}
```

Comment:
I think POST and PATCH are used by addWorkload and updateWorkload

Reply:
You're right. They are _technically_ used, but if I recall correctly, because we are using a "dummy" RMD pod, we don't expect a legitimate response from PATCH or POST requests. We just send the request and assume it lands successfully so it doesn't affect our existing test cases.

We _could_ (if you're feeling brave!) introduce some test cases where we fail some of these requests on purpose and then test how they are handled accordingly. But to be honest I think that's probably overkill - maybe something to come back to if you have (a lot of 🤣) free time.

Reply:
Ah grand, I'll take them out so 😂